src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -36,7 +36,7 @@ public static boolean isValidTagName(String test) {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Tag // instanceof handles nulls
-                && tagName.equals(((Tag) other).tagName)); // state check
+                && tagName.equalsIgnoreCase(((Tag) other).tagName)); // state check
```

Comment:
Should Tags object ignore case?
I think there is value in tags with different capitalization being different from each other.
Because of the alphanumeric constraint on tags, camel or snake case might be used to distinguish words and have different meanings
eg. WatchSuperBowl vs WatchSuperbOwl

search can and should have case sensitivity as an option, but I disagree that removing Tag object case sensitivity is the right implementation

Reply:
I agree with what @cookiedan42 said. 

On another note, I think it might be easier to follow the code you break this line into an if-else chain.

Reply:
Hm ok true.
Perhaps we can make it such that by default tag search is case insensitive then with the flags, users can make it case sensitive. For now, I'll delete this and use another function to compare tags when searching. 
Thanks for comments!